### DISCUSSION TITLE
Lesson 5: unable to add account from .env

### DISCUSSION THREAD

**Discussion by cromewar** [smartcontractkit/full-blockchain-solidity-course-py#123] [Q&A] [answered]
Hello there I get a KeyError when trying to import the key from .env

![image](https://user-images.githubusercontent.com/20199180/135464938-046f14c3-04a8-424d-bdc3-8b6bb15599aa.png)

This is my code:

from brownie import accounts, config

```python
def deploy_simple_storage():
    # account = accounts[0]
    # print(account)
    # account = accounts.load("crome-account")
    # print(account)
    account = accounts.add(config["wallets"]["from_key"])
    print(account)


def main():
    deploy_simple_storage()
```

brownie-config.yaml file:
```python
dotenv: .env
wallets:
  from_key: ${PRIVATE_KEY}

```
I have no idea why is not working, any help is appreciated.

**Comment by cromewar** [comment]
Aditionally I had been reading the documentation when I noticed something. According to documentation there is not a parameter called "wallets" on the brownie-config.yaml which makes all even more confusing.
Link to Docs: https://eth-brownie.readthedocs.io/en/stable/config.html

**Comment by cromewar** [comment, marked as answer]
I solved it by changing the folder structure.

**Reply by PatrickAlphaC** [reply]
Nice work! Thanks for posting and following up.

**Reply by KennethJChew** [reply]
What changes did you make to the folder structure? I'm having a similar issue when im getting a TypeError: string indices must be integers

**Reply by cromewar** [reply]
> What changes did you make to the folder structure? I'm having a similar issue when im getting a TypeError: string indices must be integers

Well I had two folder one for the virtual env and another for the project, I had brownie-config.yaml on the same path of bot folder, I just moved it to the project folder and the problem was solved.

**Reply by JamiDEV** [reply]
Can post a screen shot of what you mean I'm haveing the same error but I dont understand

**Reply by cromewar** [reply]
Basically I had brownie config outside the project folder, I just moved it and it worked fine, also @JamiDEV  remember the name should be `brownie-config.yaml`.

**Reply by JamiDEV** [reply]
I do as well, where did you put it? As a matter of fact, you are correct about the project structure.

**Reply by cromewar** [reply]
On the project folder like this:

![image](https://user-images.githubusercontent.com/20199180/140963734-4299b58c-2c38-4ab2-a434-89f43c6e1777.png)

**Reply by JamiDEV** [reply]
this is what I have

![Screen Shot 2021-11-09 at 9 29 25 AM](https://user-images.githubusercontent.com/53233466/140964663-67581052-95da-4928-a932-c5f3694464b4.png)

**Reply by cromewar** [reply]
delete the build folder and compile again.

**Reply by JamiDEV** [reply]
I'll give a shot, thank you

**Reply by JamiDEV** [reply]
so it didn't work, I'll keep looking for solutions

**Reply by mohammadbadshah** [reply]
Hey @JamiDEV did find any workaround?

**Reply by cromewar** [reply]
Could you pals share for brownie-config.yaml?

**Reply by JamiDEV** [reply]
@mohammadbadshah deleting the build folder was the solution.

**Reply by Daruma1800** [reply]
@cromewar & @jamiDEV I deleted the build folder and compiled again and still getting the same error? Please find my code along with my folders and terminal where I recompiled the build folder. Do you see anything in addition causing this? 

![image](https://user-images.githubusercontent.com/94502087/149209862-3d90cd2b-a675-4c76-bc35-414accc5a2a1.png)

**Reply by cromewar** [reply]
@Daruma1800 check your brownie-config.yaml to find any typos or something like that.

**Reply by Mattsll99** [reply]
Hi @Daruma1800! your .env and brownie-config.yaml fils are in the test folder. Just move them to the root folder.

**Comment by shlok-codes** [comment]
I am having the same issue. I tried deleting the build folder. And yes the yaml file is in the brownie_simple_storage folder. Could you advice what the issue may be?

the yaml code:
```
dotenv: .env
wallets:
  from_key: ${"PRIVATE_KEY"}
```

the deploy code:
```
from brownie import accounts, config


def deploy_simple_storage():
    # account = accounts[0]
    # print(account)
    # account = accounts.load("freecodecamp-account")
    # print(account)
    account = accounts.add(config["wallets"]["from_key"])
    print(account)


def main():
    deploy_simple_storage()
```

**Reply by shlok-codes** [reply]
actually my error specifically is:
![image](https://user-images.githubusercontent.com/30735595/147391550-9009875e-c16e-454a-845f-5c9526a9b6d3.png)

**Reply by shlok-codes** [reply]
Answered my own question. I had PRIVATE_KEY in quotes in the yaml. lol
Merry Christmas!

**Reply by PatrickAlphaC** [reply]
Merry Christmas!

**Reply by shlok-codes** [reply]
Thanks professor Patrick! Happy new year my friend!

**Comment by Daruma1800** [comment]
Thank you all, it’s been running well now!
